tests/integration/server/api/roles_api_test.py

```diff
@@ -403,57 +418,495 @@ def test_add_role_task():
+
+        # Check for role members
+        are_members_removed = wait_for_resource_removal_in_db(
+            "role_members", "role_id", role_id
+        )
+
+        assert are_members_removed is True
+
+    # clean up
+    delete_user_by_username("walt1")
+
+
+def test_delete_role_with_proposals():
+    """
+    Test the delete roll api
+
+    Create a test user for auth
+    Create a test user for role membership
+    Create a test role
+    Propose adding the second user as a member
+    Deletes the test role
+    Check that the membership proposal was autorejected
+    """
+    with requests.Session() as session:
+        # Create test user
+        role_owner = {
+            "name": "Fin Pirate",
+            "username": "fin1",
+            "password": "12345678",
+            "email": "fin@pirate.co",
+        }
+        user_response = create_test_user(session, role_owner)
+        assert user_response.status_code == 200, (
+            "Error creating user: %s" % user_response.json()
+        )
+        role_owner["next_id"] = user_response.json()["data"]["user"]["id"]
+
+        # Create test user
+        new_member = {
+            "name": "Frank Pirate",
+            "username": "frank1",
+            "password": "12345678",
+            "email": "frank@pirate.co",
+        }
+        user_response = create_test_user(session, new_member)
+        assert user_response.status_code == 200, (
+            "Error creating user: %s" % user_response.json()
+        )
+        new_member["next_id"] = user_response.json()["data"]["user"]["id"]
+
+        # Auth as new_member
+        payload = {"id": new_member["username"], "password": new_member["password"]}
+        auth_response = session.post(
+            "http://rbac-server:8000/api/authorization/", json=payload
+        )
+        assert auth_response.status_code == 200, "Failed to authenticate as %s. %s" % (
+            new_member["username"],
+            auth_response.json(),
+        )
+
+        # Create test role
+        role_resource = {
+            "name": "Men of Low Moral Fiber",
+            "owners": role_owner["next_id"],
+            "administrators": role_owner["next_id"],
+        }
+        role_response = session.post(
+            "http://rbac-server:8000/api/roles", json=role_resource
+        )
+        assert role_response.status_code == 200, (
+            "Error creating role: %s" % role_response.json()
+        )
+
+        # Wait for role in db
+        role_id = role_response.json()["data"]["id"]
+        is_role_in_db = wait_for_role_in_db(role_id)
+        assert (
+            is_role_in_db is True
+        ), "Couldn't find role in rethinkdb, maximum attempts exceeded."
+
+        # Add role member
+        role_update_payload = {
+            "id": new_member["next_id"],
+            "reason": "Integration test of membership proposal removal on role deletion.",
+            "metadata": "",
+        }
+        member_response = session.post(
```

Comment:
I only create one user in that test, so I'm not sure what you mean

Reply:
In line 746 you authenticated as the new_member user.

In line 760 you created a role with the role_owner user as the admin and the owner of the role.

In line 780, you created a add_role_member proposal while signed in as new_member user instead of the role_owner user.